### PR TITLE
Fix dropfiles type after symfony BC in 2.8.31

### PR DESCRIPTION
**Pull request by davidbeig**
The last update we did to the ^2.8 minor of symfony brought some breaking changes. This PR tries to solve them.

We found out that there were some changes in the FileType of the Symfony Forms. Now the Request Handler will check that the content of the file is an actual updated file. Since we had some tweaks going on on most of our forms, we have to change them.

We were using our micro api to handle the upload of files and the relation between them and the models they relate to. An example is the images uploaded to a project. They were being saved inside a folder with the id of the project as well as being saved in the table ```project_images```.

This logic has to be moved from the microapi to the form save or the model.

The list of forms that have been modified are:

- [X] UserProfileForm.php
- [x] AdminChannelResourceEditForm.php
- [x] AdminFootprintEditForm.php
- [x] AdminPostEditForm.php
- [x] AdminProgramForm.php
- [x] AdminSdgEditForm.php
- [x] AdminSectionForm.php
- [x] AdminSocialCommitmentEditForm.php
- [x] AdminSphereEditForm.php
- [x] AdminStoryEditForm.php
- [x] AdminWorkshopEditForm.php
- [x] ProjectPostForm.php
- [x] ProjectStoryForm.php
- [x] QuestionnaireForm.php

### CODE REVIEW

**Comment by microstudi** [comment]
@davidbeig @xarly89 necesitare que facis el review tu d'aixo el dilluns. Jo ja estaré de vacances

File: src/Goteo/Library/Forms/Admin/AdminFootprintEditForm.php [hunk 1]
```diff
@@ -110,14 +106,26 @@ public function save(FormInterface $form = null, $force_save = false) {
...
-        if($data['icon'] && is_array($data['icon'])) {
-            $data['icon'] = $data['icon'][0];
-        } else {
-            $data['icon'] = null;
-        }
+        // if($data['icon'] && is_array($data['icon'])) {
+        //     $data['icon'] = $data['icon'][0];
+        // } else {
+        //     $data['icon'] = null;
+        // }
```
Comment (carlos-mg89) [inline src/Goteo/Library/Forms/Admin/AdminFootprintEditForm.php]: @davidbeig what do you think if we get rid of this commented code? We already have code versioning to keep history of it :stuck_out_tongue:

File: src/Goteo/Library/Forms/Model/QuestionnaireCreateForm.php [hunk 2]
```diff
@@ -50,7 +50,7 @@ public function addQuestion($question)
         if ($config->attr) { $config->attr = (array) $config->attr;
         }
         if ($config->type == "dropfiles") {
-            $config->url = '/api/questionnaire/documents';
+            // $config->url = '/api/questionnaire/documents';
```
Comment (carlos-mg89) [inline src/Goteo/Library/Forms/Model/QuestionnaireCreateForm.php]: @davidbeig We could remove this line

File: src/Goteo/Library/Forms/Model/QuestionnaireForm.php [hunk 3]
```diff
@@ -53,7 +53,7 @@ public function createForm()
                 $question->vars->attr = (array) $question->vars->attr;
 
             if ($type == "dropfiles") {
-                $question->vars->url = '/api/questionnaire/documents';
+                // $question->vars->url = '/api/questionnaire/documents';
```
Comment (carlos-mg89) [inline src/Goteo/Library/Forms/Model/QuestionnaireForm.php]: @davidbeig We could remove this line

**Comment by carlos-mg89** [comment]
Reviewed forms:

- [x] UserProfileForm.php
- [x] AdminChannelResourceEditForm.php
- [x] AdminFootprintEditForm.php
- [x] AdminPostEditForm.php
- [x] AdminProgramForm.php
- [x] AdminSdgEditForm.php => Fixed! It wasn't working entirely OK.
- [x] AdminSectionForm.php
- [x] AdminSocialCommitmentEditForm.php
- [x] AdminSphereEditForm.php
- [x] AdminStoryEditForm.php => Fixed! It wasn't working entirely OK.
- [x] AdminWorkshopEditForm.php
- [x] ProjectPostForm.php => Fixed! It wasn't working entirely OK.
- [x] ProjectStoryForm.php => Fixed! It wasn't working at all.
- [x] QuestionnaireForm.php

@davidbeig here is an up-to-date list of the forms I've been able to verify.

File: composer.json [hunk 4]
```diff
@@ -13,25 +13,25 @@
...
+        "symfony/console": "^2.8",
+        "symfony/translation": "^2.8",
+        "symfony/finder": "^2.8",
+        "symfony/dom-crawler": "^2.8",
+        "symfony/css-selector": "^2.8",
+        "symfony/process": "^2.8",
+        "symfony/form": "^2.8",
+        "symfony/templating": "^2.8",
+        "symfony/framework-bundle": "^2.8",
+        "symfony/validator": "^2.8",
```
Comment (carlos-mg89) [inline composer.json:40]: @davidbeig great! However, we are missing the updated composer.lock only with the Symfony packages

Comment (davidbeig) [reply composer.json:40]: Added!

**Comment by davidbeig** [comment]
Hi @carlos-mg89 thanks for reviewing it!

I have left the paths to where you can find the forms in your goteo application inside your message.

> Reviewed forms:
> 
>     * [ ]  **UserProfileForm.php => Can add picture, but cannot remove it**
> 
>     * [x]  AdminChannelResourceEditForm.php
> 
>     * [ ]  AdminFootprintEditForm.php => /admin/categories/footprint (and then edit/create one entry)
> 
>     * [x]  AdminPostEditForm.php
> 
>     * [x]  AdminProgramForm.php
> 
>     * [ ]  AdminSdgEditForm.php => /admin/categories/sdg ( the same as in footprint )
> 
>     * [x]  AdminSectionForm.php
> 
>     * [ ]  AdminSocialCommitmentEditForm.php => /admin/categories/socialcommitment ( same )
> 
>     * [ ]  AdminSphereEditForm.php => /admin/categories/sphere (same)
>
>     * [x]  AdminStoryEditForm.php
> 
>     * [x]  AdminWorkshopEditForm.php
> 
>     * [x]  ProjectPostForm.php
> 
>     * [ ]  ProjectStoryForm.php => /dashboard/project/{$project_id}/story ( The project has to be funded)
> 
>     * [ ]  QuestionnaireForm.php => This one is quite tricky... You have to create a questionnaire and then answer it. To do so you can use a channel. But we can schedule a moment and i just show you.
> 
> 
> @davidbeig here is an up-to-date list of the forms I've been able to verify.

**Review by carlos-mg89 — APPROVED** [review]

**Review by javicarrillo — APPROVED** [review]
Great work! I think it's all ok too

**Pull request merged**